docs/development/go/go-context.md

```diff
@@ -0,0 +1,468 @@
+
+### Explaining the Go code of the Example
+
+The Go code of `simple.go` is as follows:
+
+{{< file "./simple.go" go >}}
+package main
+
+import (
+        "context"
+        "fmt"
+        "os"
+        "strconv"
+        "time"
+)
+
+func f1(t int) {
+        c1 := context.Background()
+        c1, cancel := context.WithCancel(c1)
+        defer cancel()
+
+        go func() {
+                time.Sleep(4 * time.Second)
+                cancel()
+        }()
+
+        select {
+        case <-c1.Done():
+                fmt.Println("f1():", c1.Err())
+                return
+        case r := <-time.After(time.Duration(t) * time.Second):
+                fmt.Println("f1():", r)
+        }
+        return
+}
+
+func f2(t int) {
+        c2 := context.Background()
+        c2, cancel := context.WithTimeout(c2, time.Duration(t)*time.Second)
+        defer cancel()
+
+        go func() {
+                time.Sleep(4 * time.Second)
+                cancel()
+        }()
+
+        select {
+        case <-c2.Done():
+                fmt.Println("f2():", c2.Err())
+                return
+        case r := <-time.After(time.Duration(t) * time.Second):
+                fmt.Println("f2():", r)
+        }
+        return
+}
+
+func f3(t int) {
+        c3 := context.Background()
+        deadline := time.Now().Add(time.Duration(2*t) * time.Second)
+        c3, cancel := context.WithDeadline(c3, deadline)
+        defer cancel()
+
+        go func() {
+                time.Sleep(4 * time.Second)
+                cancel()
+        }()
+
+        select {
+        case <-c3.Done():
+                fmt.Println("f3():", c3.Err())
+                return
+        case r := <-time.After(time.Duration(t) * time.Second):
+                fmt.Println("f3():", r)
+        }
+        return
+}
+
+func main() {
+        if len(os.Args) != 2 {
+                fmt.Println("Need a delay!")
+                return
+        }
+
+        delay, err := strconv.Atoi(os.Args[1])
+        if err != nil {
+                fmt.Println(err)
+                return
+        }
+        fmt.Println("Delay:", delay)
+
+        f1(delay)
+        f2(delay)
+        f3(delay)
+}
+{{< /file >}}
+
+The program contains four functions including the `main()` function. Functions `f1()`,
+`f2()` and `f3()` require just one parameter, which is a time delay, because everything
+else is defined inside the function.
+
+The first thing to remember is that you need to call the `context.Background()` function
+in order to initialize an empty `Context`. The other function that can create an empty
+`Context` is `context.TODO()` and is going to be presented later in this guide.
+
+Notice that the `cancel` variable in `f1()`, which is a function, is one of the return values of
+`context.CancelFunc()`. The `context.WithCancel()` function uses an existing `Context` and creates a
+child of it with cancellation. The `context.WithCancel()` function also retuns a `Done` channel
```

Comment:
Explaining the child part here is good, and would also be good to do where the documentation from the `context` source was included.

Reply:
Did that in the `context` introduction.